src/components/TransactionConfirmationModal/index.tsx

```diff
@@ -63,15 +64,35 @@ export function ConfirmationPendingContent({ onDismiss, pendingText }: { onDismi
 }
 
 function TransactionSubmittedContent({
+  trade,
   onDismiss,
   chainId,
   hash,
 }: {
+  trade?: Trade
   onDismiss: () => void
   hash: string | undefined
   chainId: ChainId
 }) {
   const theme = useTheme()
+  const { t } = useTranslation()
+
+  const isGnosisProtocolTrade = trade instanceof GnosisProtocolTrade
+  const link =
+    chainId &&
+    hash &&
+    (isGnosisProtocolTrade
+      ? getGnosisProtocolExplorerOrderLink(chainId, hash)
+      : getExplorerLink(chainId, hash, 'transaction'))
+
+  const externalLinkText = `${isGnosisProtocolTrade ? t('viewOnCowExplorer') : t('viewOnBlockExplorer')}`
+  const explorerExternalLink = chainId && hash && (
```

Comment:
```suggestion
  const explorerExternalLink = chainId && hash && (
```

&& check is tricky make explicit checks

```
chainId && hash !== undefiend ? <ExternalLink href={link as string}>
      <Text fontWeight={500} fontSize="13px">
        {externalLinkText}
      </Text>
    </ExternalLink> : null
```